processing/src/main/java/io/druid/query/MetricsEmittingExecutorService.java

```diff
@@ -66,6 +71,7 @@ public void emitMetrics(ServiceEmitter emitter, ServiceMetricEvent.Builder metri
   {
     if (delegate instanceof PrioritizedExecutorService) {
       emitter.emit(metricBuilder.build("segment/scan/pending", ((PrioritizedExecutorService) delegate).getQueueSize()));
+      emitter.emit(metricBuilder.build("merge/buffers/size", ((DefaultBlockingPool) mergeBufferPool).getPoolSize()));
```

Comment:
`getPoolSize()` is not the right method to use here. It's the number of buffers in the pool that have _not_ been handed out, but we want this metric to be the number that _are_ handed out. Try adding a new method to the BlockingPool interface like `getUsedBufferCount()` and then implementing that in DefaultBlockingPool.

Reply:
Done.